src/components/Payroll/statusConfig.ts

```diff
@@ -0,0 +1,293 @@
+export type PayrollStatusBadge = {
+  label?: string
+  variant: 'success' | 'warning' | 'error' | 'info'
+  translationKey: string
+  translationParams?: Record<string, string | number>
```

Comment:
I'm wondering if this could be resolved by just passing the translation itself in and having a children prop instead of needing to have this put together the key and params